pennylane/templates/subroutines/multiplier.py

```diff
@@ -248,13 +229,27 @@ def compute_decomposition(
             work_wire_aux = ()
             wires_aux = work_wires[: len(x_wires)]
             wires_aux_swap = wires_aux
-        op_list.extend(_mul_out_k_mod(k, x_wires, mod, work_wire_aux, wires_aux))
-        for x_wire, aux_wire in zip(x_wires, wires_aux_swap):
-            op_list.append(SWAP(wires=[x_wire, aux_wire]))
+
+        compute_op = change_op_basis(
+            QFT(wires=wires_aux),
+            ControlledSequence(PhaseAdder(k, wires_aux, mod, work_wire_aux), control=x_wires),
+        )
+
+        target_op = prod(*reversed([SWAP(wires) for wires in zip(x_wires, wires_aux_swap)]))
+
         inv_k = pow(k, -1, mod)
+        uncompute_op = change_op_basis(
+            QFT(wires=wires_aux),
+            adjoint(
+                ControlledSequence(
+                    PhaseAdder(inv_k, wires_aux, mod, work_wire_aux), control=x_wires
+                )
+            ),
+        )
 
-        for op in reversed(_mul_out_k_mod(inv_k, x_wires, mod, work_wire_aux, wires_aux)):
-            op_list.append(adjoint(op))
+        op_list = [
+            change_op_basis(compute_op, target_op, uncompute_op),
+        ]
```

Comment:
black doing black things...
```suggestion
        op_list = [change_op_basis(compute_op, target_op, uncompute_op)]
```